mm2src/coins/nft.rs

```diff
@@ -1396,3 +1372,44 @@ pub(crate) fn get_domain_from_url(url: Option<&str>) -> Option<String> {
     url.and_then(|uri| Url::parse(uri).ok())
         .and_then(|url| url.domain().map(String::from))
 }
+
+/// Clears NFT data from the database for specified chains.
+pub async fn clear_nft_db(ctx: MmArc, req: ClearNftDbReq) -> MmResult<(), ClearNftDbError> {
+    if req.clear_all {
+        let nft_ctx = NftCtx::from_ctx(&ctx).map_to_mm(ClearNftDbError::Internal)?;
+        let storage = nft_ctx.lock_db().await?;
+        storage.clear_all_nft_data().await?;
+        storage.clear_all_history_data().await?;
+    } else if !req.chains.is_empty() {
+        let nft_ctx = NftCtx::from_ctx(&ctx).map_to_mm(ClearNftDbError::Internal)?;
+        let storage = nft_ctx.lock_db().await?;
+        for chain in req.chains.iter() {
+            if let Err(e) = clear_data_for_chain(&storage, chain).await {
+                error!("Failed to clear data for chain {}: {}", chain, e);
+            }
+        }
+    } else {
+        return MmError::err(ClearNftDbError::InvalidRequest(
+            "Nothing to clear was specified".to_string(),
+        ));
+    }
+
+    Ok(())
+}
```

Comment:
Hmm I had almost the same implementation first time, but got [review suggestion](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2039#discussion_r1447133062). [commit with changes](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2039/commits/d505c5619df67aeb58e8b494847b0c2d76e63f05)

Perhaps current version doesnt look beautiful, but we avoid unnecessary storage creation.
Personally, Im really fine with both variants. 
However, I see that your version is more readable and maintainable. Also, I like that you're collecting all errors in a vector, instead of just showing them in logs.

I suppose I should return to the first implementation, following your approach.

Reply:
I like to follow early error return pattern when possible. It usually leads to clear logic and less nesting.

Reply:
We can move this part
```rust
    let nft_ctx = NftCtx::from_ctx(&ctx).map_to_mm(ClearNftDbError::Internal)?;
    let storage = nft_ctx.lock_db().await?;

    if req.clear_all {
        storage.clear_all_nft_data().await?;
        storage.clear_all_history_data().await?;
        return Ok(());
    }
```
to the top to not check `!req.clear_all` again, but we will lock the db for no reason if `Nothing to clear was specified ` error was to occur. Also in the current code if `req.clear_all` is false we do another check for `!req.chains.is_empty()` so we still check req multiple times.

Reply:
> Maybe this is the best version

Love it, thanks! 
I see, we should execute `clear_all` first, if its true. Bcz in non wasm target even if `nft list / transfer history` tables dont exist, `scanned_nft_blocks` table can be dropped only with `clear_all` flag.

> I like to follow early errors return pattern when possible. It usually leads to clear logic/readability and less nesting.

Noted